File: frontend/src/services/diaryEntryService.js
import apiClient  from "./apiClient";

const diaryEntryService = {
    createDiaryEntry: (description, images, mood, selectedHabits) => {
        return apiClient.post('/diary-entry', {description: description, images: images, mood: mood, habits: selectedHabits});
    },

    getDiaryEntries: (page, size) => {
        return apiClient.get(`/diary-entry?page=${page}&size=${size}`);
    },
    getAllMoods: () => {
        return apiClient.get('/diary-entry/moods');
    },
    getLatestDiaryEntry: () => {
        return apiClient.get(`/diary-entry/latest`);
    }
}

export default diaryEntryService;   

File: backend/src/main/java/com/tfg/tfg_app/model/services/DiaryEntryService.java
package com.tfg.tfg_app.model.services;

import java.time.LocalDateTime;
import java.util.List;

import org.springframework.data.domain.Page;
import com.tfg.tfg_app.model.common.exceptions.InstanceNotFoundException;
import com.tfg.tfg_app.model.entities.DiaryEntry;
import com.tfg.tfg_app.model.entities.Mood;
import com.tfg.tfg_app.model.entities.UserHabit;
import com.tfg.tfg_app.model.services.exceptions.DuplicatedEntryException;

public interface DiaryEntryService {

    DiaryEntry createDiaryEntry(Long userId, DiaryEntry diaryEntry, List<byte[]> images, List<UserHabit> habits) throws DuplicatedEntryException, InstanceNotFoundException;

    DiaryEntry getDiaryEntryById(Long id) throws InstanceNotFoundException;

    Page<DiaryEntry> getDiaryEntriesByUserId(Long userId, int page, int size);

    List<Mood> getAllMoods();

    DiaryEntry getWeeksMostFrequentMood(Long userId, LocalDateTime startDate, LocalDateTime endDate);

    List<DiaryEntry> getDiaryEntriesByUserIdAndDate(Long userId, LocalDateTime startDate, LocalDateTime endDate);

    DiaryEntry getLatestDiaryEntry(Long userId) throws InstanceNotFoundException;
}
